### PR TITLE
registration UI progress

### PR DESCRIPTION
**Pull request by bluquar**
https://user-images.githubusercontent.com/4309735/143270883-8808e8e6-0d01-49e8-9c06-d440f8541355.mov

### CODE REVIEW

File: screens/RegisterScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  React.useEffect(() => {
+    if (accessTokenAfterRegistering != null) {
+      PersistentClientStorage.set('token', accessTokenAfterRegistering);
+    }
+    // Navigate to "Activate Account" screen
+  }, [accessTokenAfterRegistering]);
+
+  // Email addresses can't be truly validated -- https://tinyl.io/5Cje
+  // Be permissive but make sure it at least has an @ sign
+  const isEmailValid = email.includes('@');
```
Comment (lowellbander) [inline screens/RegisterScreen.tsx:40]: hehe

**Review by lowellbander — APPROVED** [review]
holy fuck u r a rockstar!!!!

looking at the test plan video, it looks like you can log in even without verifying your email address. Is that right? If so, that's probably OK for very early users but dangerous otherwise, because people can potentially impersonate others, right?

**Comment by bluquar** [comment]
Yes, that’s right. I’m thinking of blocking access to anything except the
verification screen if your account hasn’t been verified. But I don’t quite
understand the risk — how could someone impersonate someone else? Email
addresses are not visible to other users.

**Comment by bluquar** [comment]
First thing that comes to mind seems not that serious, but: what happens if
I use your email to create an account and choose a username to go with it -
- even though I can't verify the account, does that mean your email address
is now bound to a username that I chose?

Hmm wait, we're not gonna have usernames, right? Just email, public name,
and nickname. Yeah not sure what the risk is here.

What if I choose a password to go with your email. If we don't have
password reset enabled, then you won't be able to login because I'm the one
who chose the password for your email, right?

**Comment by bluquar** [comment]
We'll have password reset enabled. But yeah to avoid that scenario -- which
would essentially mean one person takes over another person's account (even
though it is associated with the new person's email address) -- we could
just require people to verify their email in order to be considered "logged
in."

**Pull request merged**